modules/integration/src/test/scala/scala/cli/integration/RunTestDefinitions.scala

```diff
@@ -1916,4 +1958,41 @@ abstract class RunTestDefinitions(val scalaVersionOpt: Option[String])
       expect(mainClasses == Set(scalaFile1, scalaFile2, s"$scriptsDir.${scriptName}_sc"))
     }
   }
+
+  test("correctly run a script snippet") {
+    emptyInputs.fromRoot { root =>
+      val msg = "123456" // FIXME: change this to a a non-numeric string when Windows encoding is handled properly
```

Comment:
the reason for this is that when code is being passed from a Windows terminal, Windows encoding is being used, while our inputs logic assumes everything is always to be `UTF-8`.
Fixing this requires a bit of a refactor, so I'd rather do it in a separate PR.

Reply:
(so as a result, for now this feature will be buggy on Windows)